src/hooks/useSpawnEngine.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+
+import { spawnEngine } from '@/engine/spawn'
+
+import { useOnce } from './useOnce'
+
+export function useSpawnEngine() {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const spawnAsync = () => {
```

Comment:
(Nitpick: we could do without the wrapper)

Reply:
Actually I didn't manage to. Because it is async yet must be passed as synchronous code to `useEffect`.

I will merge like this for the sake of starting the work on other tasks. Feel free to open a PR if you have an alternative 🙂